#VRML_SIM R2019b utf8
WorldInfo {
  info [
    "Example world for demonstrating the InertialUnit."
  ]
  title "Leg Robot"
  basicTimeStep 10
  FPS 100
}
Viewpoint {
  orientation -0.9680095269861047 0.1598232979658135 0.19342716740796453 0.17978997878661898
  position 0.83479737407051 -2.1006089739090466 1.5163861923789224
}
TexturedBackground {
  texture "noon_building_overcast"
}
TexturedBackgroundLight {
  texture "noon_building_overcast"
}
StraightStairs {
  translation 0 -3.03 0
  leftRail []
  rightRail []
}
DEF Leg Robot {
  translation 0.7773860517510334 -2.5812960578719086 -2.4846326139497996e-09
  rotation 0.0008082307240439361 0.9999996733814952 -1.968385989368622e-09 3.1415789993268093
  children [
    DEF FOOT Solid {
      rotation 0 0 1 0
      children [
        DEF ANKLE Transform {
          translation -0.025 0.017 0
          children [
            DEF ANKLE_X SliderJoint {
              jointParameters JointParameters {
                position -1.5772624760841772e-07
                axis 1 0 0
              }
              device [
                DEF ANKLE_X LinearMotor {
                  name "ankle x linear motor"
                  maxVelocity 10000
                  maxForce 10000
                }
              ]
              endPoint DEF ANKLE_X Solid {
                translation -1.6202595417391176e-07 0.00046072387116100066 -5.90185618959185e-17
                rotation 2.9504199197464706e-08 -8.923450162217335e-11 -0.9999999999999996 -5.047846092305859e-07
                children [
                  DEF ANKLE_X Shape {
                    appearance PBRAppearance {
                      baseColor 0 0.5 0
                    }
                    geometry Box {
                      size 0.1 0.001 0.1
                    }
                  }
                  DEF ANKLE_Y SliderJoint {
                    jointParameters JointParameters {
                      position -9.815876311548952e-05
                      axis 0 1 0
                    }
                    device [
                      DEF ANKLE_Y LinearMotor {
                        name "ankle y linear motor"
                        maxVelocity 10000
                        maxForce 10000
                      }
                    ]
                    endPoint DEF ANKLE_Y Solid {
                      translation 0 0.0009018608564882438 0
                      children [
                        DEF ANKLE_Y Shape {
                          appearance PBRAppearance {
                            baseColor 0.5 0 0
                          }
                          geometry Box {
                            size 0.1 0.001 0.1
                          }
                        }
                        DEF ANKLE_Z SliderJoint {
                          jointParameters JointParameters {
                            position 2.592186515914435e-12
                          }
                          device [
                            DEF ANKLE_Z LinearMotor {
                              name "ankle z linear motor"
                              maxVelocity 10000
                              maxForce 10000
                            }
                          ]
                          endPoint DEF ANKLE_Z Solid {
                            translation 0 0.001 2.592186515914435e-12
                            children [
                              DEF ANKLE_Z_Slider Shape {
                                appearance PBRAppearance {
                                  baseColor 0 0 0.5
                                }
                                geometry Box {
                                  size 0.1 0.001 0.1
                                }
                              }
                              DEF ANKLE_Z HingeJoint {
                                jointParameters HingeJointParameters {
                                  position 4.344822151844877e-08
                                  axis 0 0 1
                                  anchor 0 0.0155 0
                                }
                                device [
                                  PositionSensor {
                                    name "ankle z position sensor"
                                  }
                                  RotationalMotor {
                                    name "ankle z rotational motor"
                                    maxVelocity 10000
                                    maxTorque 10000
                                  }
                                ]
                                endPoint DEF SHANK Solid {
                                  translation 2.1858099999999976e-10 0.015500000000000005 0
                                  rotation 0 0 1 4.2146848510894035e-08
                                  children [
                                    DEF ANKLE_SHANK Group {
                                      children [
                                        DEF COM Transform {
                                          translation 0 0.4 0
                                          children [
                                            DEF COM Shape {
                                              appearance PBRAppearance {
                                                baseColor 0 1 1
                                              }
                                              geometry Sphere {
                                                radius 0.1
                                              }
                                            }
                                          ]
                                        }
                                        DEF ANKLE_J Transform {
                                          rotation 1 0 0 -1.57
                                          children [
                                            Shape {
                                              appearance PBRAppearance {
                                                baseColor 0.5 0.5 0
                                              }
                                              geometry Cylinder {
                                                height 0.1
                                                radius 0.015
                                              }
                                            }
                                          ]
                                        }
                                        DEF SHANK Transform {
                                          translation 0 0.2 0
                                          children [
                                            DEF SHANK Shape {
                                              appearance PBRAppearance {
                                                baseColor 0.5 0 0
                                              }
                                              geometry Box {
                                                size 0.03 0.4 0.1
                                              }
                                            }
                                          ]
                                        }
                                      ]
                                    }
                                  ]
                                  boundingObject USE ANKLE_SHANK
                                  physics Physics {
                                    density -1
                                    mass 5
                                  }
                                }
                              }
                            ]
                            boundingObject USE ANKLE_Z_Slider
                            physics Physics {
                              density -1
                              mass 0.001
                            }
                          }
                        }
                      ]
                      boundingObject USE ANKLE_Y
                      physics Physics {
                        density -1
                        mass 0.001
                      }
                    }
                  }
                ]
                name "ankle x solid"
                boundingObject USE ANKLE_X
                physics Physics {
                  density -1
                  mass 0.001
                }
              }
            }
          ]
        }
        DEF FOOT TouchSensor {
          translation 3.8971943472354553e-07 0.002715976620534466 -3.5357664231628217e-12
          rotation 5.667726831411673e-07 1.3874896403731003e-08 0.9999999999998394 -6.624313494034713e-06
          name "foot touch sensor"
          boundingObject Shape {
            geometry Box {
              size 0.25 0.005 0.1
            }
          }
          physics Physics {
            density -1
            mass 0.001
          }
          rotationStep 0.261799
          recognitionColors [
            0 0 0
          ]
          type "force-3d"
          lookupTable [
            0 0 0
            5000 5000 0
          ]
        }
        DEF Foot Group {
          children [
            Transform {
              translation 0 0.011 0
              children [
                Shape {
                  appearance DEF METAL_APPEARANCE PBRAppearance {
                    baseColor 0.5 0.5 0.5
                    roughness 1.1102230246251565e-16
                    metalness 0
                  }
                  geometry Box {
                    size 0.25 0.012 0.1
                  }
                }
              ]
            }
          ]
        }
      ]
      name "foot solid"
      boundingObject USE Foot
      physics Physics {
        density -1
        mass 0.5
      }
    }
  ]
  name "leg"
  physics Physics {
    density -1
    mass 0.001
  }
  rotationStep 0.261799
  controller "ZMP_control"
}
